common/common/src/main/java/io/helidon/common/SerializationConfig.java

```diff
@@ -0,0 +1,624 @@
+            String property = System.getProperty(sysProp);
+            if (property == null) {
+                return defaultValue;
+            }
+            try {
+                return Action.valueOf(property.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                List<String> validActions = Arrays.stream(Action.values())
+                        .map(Action::toString)
+                        .map(String::toLowerCase)
+                        .collect(Collectors.toList());
+
+                LOGGER.warning("System property \"" + sysProp + "\" is configured to \"" + property + "\", which is"
+                                       + " not a valid Action. Valid actions: " + validActions
+                                       + ". Using: " + defaultValue.toString().toLowerCase());
+
+                return defaultValue;
+            }
+        }
+
+        private static TraceOption configuredTrace(TraceOption defaultValue) {
+            String property = System.getProperty(PROP_TRACE);
+            if (property == null) {
+                return defaultValue;
+            }
+            try {
+                return TraceOption.valueOf(property.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                List<String> validTraceOptions = Arrays.stream(TraceOption.values())
+                        .map(TraceOption::toString)
+                        .map(String::toLowerCase)
+                        .collect(Collectors.toList());
+
+                LOGGER.warning("System property \"" + PROP_TRACE + "\" is configured to \"" + property + "\", which is"
+                                       + " not a valid TraceOption. Valid trace options: " + validTraceOptions
+                                       + ". Using: " + defaultValue.toString().toLowerCase());
+
+                return defaultValue;
+            }
+        }
+
+        @Override
+        public SerializationConfig build() {
+            this.filterPattern = getPattern();
+            return new SerializationConfig(this);
+        }
+
+        /**
+         * What action to do in case of wrong configuration of the global filter.
+         *
+         * @param onWrongConfig action to do
+         * @return updated builder
+         */
+        public Builder onWrongConfig(Action onWrongConfig) {
+            this.onWrongConfig = onWrongConfig;
+            return this;
+        }
+
+        /**
+         * What action to do in case of no configuration of the global filter.
+         *
+         * @param onNoConfig action to do
+         * @return updated builder
+         */
+        public Builder onNoConfig(Action onNoConfig) {
+            this.onNoConfig = onNoConfig;
+            return this;
+        }
+
+        /**
+         * Filter pattern to use.
+         *
+         * @param filterPattern filter pattern
+         * @return updated builder
+         */
+        public Builder filterPattern(String filterPattern) {
```

Comment:
Is `!*` always appended to this pattern? If so then I think it's more of an allow-list then a general filter. So maybe this method should be `allowPattern` or `allowListPattern`.

Reply:
This can get either a full pattern or allow list. This is not limited to allow...

Reply:
I have updated the documentation and code to support patterns with reject list